server/blob/blob-deduplicating/src/main/scala/GC.scala

```diff
@@ -129,14 +129,14 @@ case class State(blobs: Set[BlobId],
   }
   }
 
-  private def addElement[T <: Event](collection: Map[Generation, Set[T]], e: T): Map[Generation, Set[T]] = {
-    collection.updatedWith(e.generation)(previous => Some(previous.getOrElse(Set()) + e))
+  private def addElement[T <: Event](collection: Map[Generation, Seq[T]], e: T): Map[Generation, Seq[T]] = {
+    collection.updatedWith(e.generation)(previous => Some(e +: previous.getOrElse(Seq())))
   }
 }
 
 case class StabilizedState(blobs: Set[BlobId],
```

Comment:
What is the diference between state ad stabilized state?

Reply:
A `State` is used for construction, while `StabilizedState` is used for planning.

Reply:
I don't understand the disctinction. Can you explain more?

Reply:
A `State` is used during a construction phase, when all the events are played one after the other, once their all played, the last set is used to build a `Stabilized` which will be used by the planner.

There is two mains concerns:

 - `StabilizedState` have to track the blobs' references for read purposes while `State` does not, since it's write only
 -`StabilizedState` could (will) by built from other things (like Cassandra queries).

Reply:
Thanks